pkg/crc/preflight/preflight_linux.go

```diff
@@ -106,18 +127,91 @@ var libvirtPreflightChecks = [...]Check{
 		cleanupDescription: "Removing /etc/NetworkManager/dnsmasq.d/crc.conf file",
 		cleanup:            removeCrcDnsmasqConfigFile,
 	},
-	{
-		cleanupDescription: "Removing the crc VM if exists",
-		cleanup:            removeCrcVM,
-		flags:              CleanUpOnly,
-	},
 }
 
-func getPreflightChecks(experimentalFeatures bool) []Check {
-	checks := []Check{}
+var vsockPreflightChecks = Check{
+	configKeySuffix:  "check-vsock",
+	checkDescription: "Checking if vsock is correctly configured",
+	check:            checkVsock,
+	fixDescription:   "Checking if vsock is correctly configured",
+	fix:              fixVsock,
+}
+
+func checkVsock() error {
+	binary, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	getcap, err := exec.Command("getcap", binary).Output()
+	if err != nil {
+		return err
+	}
+	if !strings.Contains(string(getcap), "cap_net_bind_service+eip") {
+		return fmt.Errorf("capabilities are not correct for %s", binary)
+	}
+	info, err := os.Stat("/dev/vsock")
+	if err != nil {
```

Comment:
Fwiw, I would put the `getcap` check and the `/dev/vsock` check in 2 separate preflights. One might want to avoid one of the tests while keeping the other.

Reply:
The checks should also be in `preflight_checks_linux.go` even if the separation is really artificial. When done as part of `crc start`, this test happens too late as the daemon is already supposed to be running at that point, so even `crc setup && crc start` won't fix the issue, even if the preflight will nolonger fail.